assets/src/block-editor/QueryBlockExtension/QueryBlockExtension.js

```diff
@@ -1,154 +1,125 @@
-              for (const group of document.querySelectorAll('.components-toolbar-group')) {
-                // Find the views group from toolbar
-                if (group.innerHTML.includes('List view') || group.innerHTML.includes('Grid view')) {
-                  viewsGroup = group;
-
-                  // Remove the list view for Actions List block
-                  if (selectedBlock.attributes.namespace && selectedBlock.attributes.namespace.includes('planet4-blocks/actions-list')) {
-                    for (const children of group.children) {
-                      if (children.ariaLabel.includes('List view')) {
-                        group.removeChild(children);
-                      }
+        if (props.attributes.namespace === 'planet4-blocks/actions-list' || targetBlocks.includes(props.name)) {
+          const [isPostTemplate, setPostTemplate] = useState();
+          const [selectedBlock, setSelectedBlock] = useState();
+          const {attributes, setAttributes} = props;
+
+          const onChangeHandler = useCallback(type => {
+            const {className, query} = attributes;
+
+            const pattern = 'is-custom-layout-';
+            const layoutType = layoutTypes.find(t => t.value === type);
+
+            if(layoutType) {
+              setAttributes({
+                layout: {
+                  type: layoutType.value,
+                  columnCount: layoutType.columnCount,
+                },
+                query: {
+                  ...query,
+                  perPage: layoutType.columnCount,
+                },
+                className: ((className.includes(pattern)) ?
+                  className.replace(/\is-custom-layout-.*/, `${pattern}${layoutType.label}`) :
+                  `${className} ${pattern}${layoutType.label}`).toLowerCase(),
+              });
+            }
+          }, [attributes]);
+
+          useEffect(() => {
+            if(isPostTemplate) {
+              const timeout = setTimeout(() => {
+                  // Find all toolbar groups only when the block is selected
+                  for (const group of document.querySelectorAll('.components-toolbar-group')) {
+                    // Find the views group from toolbar
+                    if (group.innerHTML.includes('List view') || group.innerHTML.includes('Grid view')) {
+                      group.style.display = 'none';
                     }
                   }
-                }
-
-                // Move the carousel button to the views group if exists
-                if (carouselBtn && group.contains(carouselBtn) && viewsGroup) {
-                  viewsGroup.append(group);
-                }
+                }, 10);
+
+                return () => {
+                  clearTimeout(timeout);
+                };
+            }
+          }, [isPostTemplate]);
+
+          useEffect(() => {
+            if(selectedBlock && selectedBlock.name === 'core/post-template') {
```

Comment:
We need this to make sure that the `core/post-template` it is a `children` of `planet4-blocks/actions-list`.